csrc/scheduler/matmul_utils.cpp

```diff
@@ -535,7 +535,23 @@ std::shared_ptr<MatmulParams> getMatmulHeuristics(
   // Set kernel index mode
   params->cparams.index_type = getIndexType(problem_shape.value());
 
-  if (isDebugDumpEnabled(DebugDumpOption::MatmulChecks)) {
+  // Set split k using input tensor size M,N,K
+  const auto& MNK = problem_shape.value();
```

Comment:
A minor suggestion:

How about outlining splitK code to a separate function - it can accept as input reference to `params`, `problem_shape` and `device_prop`?

A side note - unconditional printing at the end of change.

Reply:
moved to a seperate function. Deleted printing used for debug.